### PR TITLE
Pass test pipelines ARTIFACTORY_REPO

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Test Pipelines require this or they default to
  sys-rt-generic-local which is undesirable.
- Add Jenkins Domain to Artifactory path
- Test Pipelines already have the domain as
  part of the path. This will match that behaviour.
- See openjdk-tests/JenkinsfileBase#L478

Issue #2637
Related #4705
[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
@jdekonin for review

**Comment by AdamBrousseau** [comment]
Test Build running
https://ci.eclipse.org/openj9/view/Pipelines/job/Pipeline-Build-Test-JDK11-linux_x86-64/675

File: buildenv/jenkins/common/build [hunk 1]
```diff
@@ -308,7 +308,7 @@ def archive() {
                     def server = Artifactory.server ARTIFACTORY_SERVER
                     // set connection timeout to 10 mins to avoid timeout on slow platforms
                     server.connection.timeout = 600
-                    def artifactory_upload_dir = "${ARTIFACTORY_REPO}/${JOB_NAME}/${BUILD_ID}/"
+                    def artifactory_upload_dir = "${ARTIFACTORY_REPO}/${JENKINS_DOMAIN}${JOB_NAME}/${BUILD_ID}/"
```
Comment (jdekonin) [inline buildenv/jenkins/common/build]: Why are we adding an additional directory layer?  Assuming this is build on #4705, could you not just override artifactory_repo with a different value?

**Comment by AdamBrousseau** [comment]
After discussion with @jdekonin and @llxia we have decided to generate the test jobs with the default values and no longer pass and Artifactory details to the test pipes. Will open a new PR to remove ARTIFACTORY_REPO.

**Pull request closed without merging**